src/ripple/app/ledger/Ledger.test.cpp

```diff
@@ -23,292 +23,493 @@
-        newLCL->updateSkipList();
-        newLCL->setClosed();
-        int asf = newLCL->peekAccountStateMap()->flushDirty(hotACCOUNT_NODE,
-                                                        newLCL->getLedgerSeq());
-        int tmf = newLCL->peekTransactionMap()->flushDirty(hotTRANSACTION_NODE,
-                                                        newLCL->getLedgerSeq());
-        using namespace std::chrono;
-        auto const epoch_offset = days{10957};  // 2000-01-01
-        std::uint32_t closeTime = time_point_cast<seconds>  // now
-                                         (system_clock::now()-epoch_offset).
-                                         time_since_epoch().count();
-        int CloseResolution = seconds{LEDGER_TIME_ACCURACY}.count();
-        bool closeTimeCorrect = true;
-        newLCL->setAccepted(closeTime, CloseResolution, closeTimeCorrect);
-        return newLCL;
+        return ledger->getSLE(index);
     }
 
-    void test_genesisLedger ()
+    std::uint32_t
+    ledger_accept()
     {
-        std::uint64_t const xrp = std::mega::num;
+        close_and_advance_by_ref(ledger, LCL);
+        return ledger->getLedgerSeq();
+    };
 
-        // Create master account
-        auto master = createAccount();
+    /* Something about something */
+    TestAccount rootAccount;
+    Ledger::pointer LCL;
+    Ledger::pointer ledger;
 
+    TestContext () :
+        rootAccount(createAccount("masterpassphrase")),
         // Create genesis ledger
-        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
-
+        LCL(createGenesisLedger(100000000000000000ull, rootAccount)),
         // Create open scratch ledger
-        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+        ledger(std::make_shared<Ledger>(false, *LCL)) {}
+};
+
+class Ledger_test : public TestSuite
+{
+    using TestAccount = TestContext::TestAccount;
+    using TXResult = TestContext::TXResult;
+
+    void
+    test_TestContext ()
+    {
+        #define TEST_ACCOUNT(name) auto name = c.createAccount(#name);
+        #define POSITION() " @ " + to_string(__FILE__) + ":" + to_string(__LINE__)
+        #define MSG(msg) "`" + to_string(#msg) + "`" + POSITION()
+        // #2 failed: `!c.makePayment(root, gw2, xrp(4000))` @ src/ripple/app/ledger/Ledger.test.cpp:392
+        #define EXPECT(expr) expect((expr), MSG(expr));
+        // #13 failed: `aliceLimit.getText() == "12"` @ src/ripple/app/ledger/Ledger.test.cpp:423
+        // Actual: 1
+        // Expected: 12
+        #define EXPECT_EQ(a, b) expectEquals(a, b, MSG(a == b));
+
+        // Stop this soab from logging crap
+        LogSeverity const sv (Logs::fromString ("fatal"));
+        auto severity = Logs::toSeverity(sv);
+        deprecatedLogs().severity(severity);
+        // TODO: restore after the test has finished
+
+        auto xrp = [](std::int64_t xrp) {return STAmount(xrp * std::mega::num);};
```

Comment:
Using an actual STAmount, gives you free + / - etc operators, and ability to << stream, so we should kill off the `Amount`